src/Views/Homepage.vala

```diff
@@ -301,4 +290,96 @@ public class AppCenter.Homepage : AbstractView {
         var apps = client.get_applications_for_category (category);
         app_list_view.add_packages (apps);
     }
+
+    private class PackageButton : Gtk.Button {
+        public AppCenterCore.Package package { get; construct; }
+
+        public PackageButton (AppCenterCore.Package package) {
+            Object (package: package);
+        }
+
+        private const string CSS = """
+            .%s {
+                background-color: %s;
+                color: %s;
+            }
+        """;
+
+        construct {
+            var name_label = new Gtk.Label (package.get_name ()) {
+                max_width_chars = 50,
+                use_markup = true,
+                wrap = true,
+                xalign = 0
+            };
+            name_label.get_style_context ().add_class (Granite.STYLE_CLASS_H1_LABEL);
+
+            var summary_label = new Gtk.Label (package.get_summary ()) {
+                max_width_chars = 50,
+                use_markup = true,
+                wrap = true,
+                xalign = 0
+            };
+            summary_label.get_style_context ().add_class (Granite.STYLE_CLASS_H3_LABEL);
+
+            string[] lines = package.get_description ().split ("\n");
+            var description = lines[0].strip ();
+
+            for (int i = 1; i < lines.length; i++) {
+                description += " " + lines[i].strip ();
+            }
+
+            int close_paragraph_index = description.index_of ("</p>", 0);
+            var description_label = new Gtk.Label (description.slice (3, close_paragraph_index)) {
+                ellipsize = Pango.EllipsizeMode.END,
+                lines = 2,
+                max_width_chars = 50,
+                use_markup = true,
+                wrap = true,
+                xalign = 0
+            };
+
+            var icon_image = new Gtk.Image.from_gicon (
+                package.get_icon (128, get_scale_factor ()),
+                Gtk.IconSize.INVALID
+            ) {
+                pixel_size = 128
+            };
+
+            var package_grid = new Gtk.Grid () {
+                column_spacing = 24,
+                halign = Gtk.Align.CENTER,
+                margin = 64,
+                valign = Gtk.Align.CENTER
+            };
+
+            package_grid.attach (icon_image, 0, 0, 1, 3);
+            package_grid.attach (name_label, 1, 0);
+            package_grid.attach (summary_label, 1, 1);
+            package_grid.attach (description_label, 1, 2);
+
+            var css_class = package.normalized_component_id.replace (".", "-");
+
+            var style_context = get_style_context ();
+            style_context.add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
```

Comment:
Why do these have suggested action?

Reply:
It was for the fallback/default instead of using custom styles and overriding things, but to get the glow (and proper outlines) it looks like I'll need to use Gtk.CSS variables anyway.